inst/stan/functions/combine_effects.stan

```diff
@@ -47,18 +47,13 @@ vector combine_effects(array[] real intercept, vector beta, matrix design,
                        vector beta_sd, matrix sd_design, int add_intercept) {
   int nobs = rows(design);
   int neffs = num_elements(beta);
-  int sds = num_elements(beta_sd);
-  vector[neffs + add_intercept] scaled_beta;
-  vector[sds + 1] ext_beta_sd;
-  ext_beta_sd[1] =  1.0;
   if (neffs) {
-    ext_beta_sd[2:(sds+1)] = beta_sd;
+    vector[num_elements(beta_sd) + 1] ext_beta_sd = append_row(1.0, beta_sd);
+    vector[nobs] result = design[:, add_intercept+1:] * (beta .* (sd_design * ext_beta_sd));
```

Comment:
I assume its coming in as 0 (false) or 1 (true), but `add_intercept` here as an `int` seems a bit code-smelly. i see stan type system doesn't support bool directly and that you can't use the constraints syntax on function arguments.

Reply:
also, this is a bit dense to read. might be preferrable to pull some of this calculation into decl of ext_beta_sd above.

maybe more like
```
scaled_beta = beta .* (sd_design * append_row(1.0, beta_sd))
result = design[:, add_intercept + 1:] * scaled_beta
```

Reply:
This got redesigned in the Sparse matrix PR./